### ISSUE TITLE
[weave-kube] too many colons in address

### ISSUE THREAD

**Issue by sdurrheimer** [weaveworks/weave#2904]
Hi there,

I deployed a new kubernetes 1.6.1 cluster via kubeadm this morning, and when applying 
 the https://git.io/weave-kube-1.6 network plugin, I face for the first time the following error:

*weave container*
```
INFO: 2017/04/20 06:29:31.586056 Command line options: map[nickname:kube-master-1 port:6783 conn-limit:30 datapath:datapath http-addr:127.0.0.1:6784 ipalloc-init:consensus=4 docker-api: ipalloc-range:10.32.0.0/12 no-dns:true status-addr:0.0.0.0:6782]
INFO: 2017/04/20 06:29:31.586115 Communication between peers is unencrypted.
INFO: 2017/04/20 06:29:31.607783 Our name is ce:7c:4d:02:b5:73(kube-master-1)
INFO: 2017/04/20 06:29:31.607804 Launch detected - using supplied peer list: [{PUBLIC-IPv4} fe80::250:56ff:fe01:18c0 192.168.0.10 fe80::20c:29ff:fe04:f34d]
INFO: 2017/04/20 06:29:31.607854 Checking for pre-existing addresses on weave bridge
INFO: 2017/04/20 06:29:31.608702 [allocator ce:7c:4d:02:b5:73] Initialising with persisted data
INFO: 2017/04/20 06:29:31.608732 Sniffing traffic on datapath (via ODP)
FATA: 2017/04/20 06:29:31.609096 too many colons in address fe80::250:56ff:fe01:18c0:0
too many colons in address fe80::20c:29ff:fe04:f34d:0
```

I successfully deployed deployed the same cluster yesterday without a problem, on the same machines.

**Comment by bboreham** [comment]
That looks like an ipv6 address, which we don't support as yet.

**Comment by sdurrheimer** [comment]
@bboreham sure, but I didn't tell weave to use IPv6 explicitly. Shouldn't weave ignore them if they are not supported ?

**Comment by bboreham** [comment]
OK, it must be coming from Kubernetes.  Yes, we should filter those out.

However another entry in that list looks a bit suspicious - `{PUBLIC-IPv4}`.  
How many nodes do you have?  
If you do `kubectl describe node kube-master-1` do you see that listed?  
If not, how about `kubectl get node kube-master-1 -o json` ?

**Comment by sdurrheimer** [comment]
My setup is 4 VMs: 1 master, 3 workers.
Each of them have 2 networks interfaces, one with a public IP (hosting provider requirement) the other with a private network.

The kubernetes cluster use the the private network for the apiserver advertise address, but will also take into account the public IP unfortunately.

**Comment by bboreham** [comment]
The Weave code specifically looks for "internal" addresses - https://github.com/weaveworks/weave/blob/master/prog/kube-peers/main.go#L39.

This is why I am interested to see the json.

**Comment by bboreham** [comment]
Oh, is that a "redacted" thing - you replaced the actual IP with "{PUBLIC-IPv4}" ?

Sorry, I'm a bit slow today.

**Comment by sdurrheimer** [comment]
yes totally. But you're right, there is something strange, my public IP is tagged as a `InternalIP`.

**Comment by tg90nor** [comment]
I have this problem as well. Single node + master, with public ipv4 and ipv6 addresses. I have redacted the ip addresses in the log below.

    INFO: 2017/04/24 16:28:59.037132 Command line options: map[conn-limit:30 ipalloc-init:consensus=4 status-addr:0.0.0.0:6782 docker-api: datapath:datapath http-addr:127.0.0.1:6784 ipalloc-range:10.32.0.0/12 nickname:kube-master.REDACTED no-dns:true port:6783]
    INFO: 2017/04/24 16:28:59.037461 Communication between peers is unencrypted.
    INFO: 2017/04/24 16:28:59.081020 Our name is f6:8e:f5:4e:c8:06(kube-master.REDACTED)
    INFO: 2017/04/24 16:28:59.081060 Launch detected - using supplied peer list: [MASTERV6 MASTERV4 NODEV6 NODEV4]
    INFO: 2017/04/24 16:28:59.081092 Checking for pre-existing addresses on weave bridge
    INFO: 2017/04/24 16:28:59.081236 weave bridge has address 10.32.0.1/12
    INFO: 2017/04/24 16:28:59.082155 [allocator f6:8e:f5:4e:c8:06] Initialising with persisted data
    INFO: 2017/04/24 16:28:59.082631 [allocator f6:8e:f5:4e:c8:06] Claim 10.32.0.1/12 for weave:expose: is in the range 10.32.0.0/12, but the allocator is not initialized yet; will try later.
    INFO: 2017/04/24 16:28:59.082653 Sniffing traffic on datapath (via ODP)
    INFO: 2017/04/24 16:28:59.129560 ->[NODEV4:6783] attempting connection
    INFO: 2017/04/24 16:28:59.130063 ->[MASTERV4:6783] attempting connection
    FATA: 2017/04/24 16:28:59.139954 too many colons in address MASTERV6:0
too many colons in address NODEV6:0

**Comment by tg90nor** [comment]
Perhaps even more curious, the node is not seeing IPv6 addresses in its peer list, and its weave-net pod is working. Only the master has the problem. I will check if this happens if I make a new cluster.

**Comment by tg90nor** [comment]
Creating a new cluster with 2 nodes, weave-net works on the master, but not on the nodes.

**Comment by marccarre** [comment]
Fixed by #2912. Merged in both `1.9` and `master`.
